src/app/file-manager/file-download-modal/file-download-modal.component.ts

```diff
@@ -92,4 +210,74 @@ export class FileDownloadModalComponent  {
       summary: warnMessage,
     });
   }
+
+  /**
+   * This function filters repeated folders when a parent
+   * is all-selected and there are sub-folders in the list,
+   * Items with indeterminate === true must be removed because
+   * selected content is included as other items.
+   * All this extra, non relevante here, items are required
+   * in the selection process.
+   */
+  filterFolderItems(sortedItems): Item[] {
+
+    const folderItems = sortedItems.filter(item => item.indeterminate === false);
+
+    const singleItems = [];
+
+    for (let i = 0; i < folderItems.length;) {
+      const parent = folderItems[i];
+
+      const isDad = this.isParent(parent, folderItems[i + 1]);
+
+      if (!singleItems.includes(parent)) {
+        singleItems.push(parent);
+      }
+
+      if (isDad) {
+        folderItems.splice(i + 1, 1);
+      } else {
+        i++;
+      }
+    }
+    return singleItems;
+  }
+
+  isParent(parent: Item, child: Item): boolean {
```

Comment:
Could be a method on `Item`, e.g. either `isChildOf()` or `isParentOf()` depending on how you want to call it.

Reply:
Ok. Is used only there but will add it to Item.